### ISSUE TITLE
getFileAsync function  unexpected bug on Word

### ISSUE THREAD

**Issue by isvdevchat** [OfficeDev/office-js#298]
Recently we encounter an issue related to Office.js API, it seems to be a bug. We may need your confirmation or advice.

We have a Word add-in which will invoke the [getFileAsync function](https://docs.microsoft.com/en-us/javascript/api/office/office.document?view=office-js#getfileasync-filetype--options--callback-) to get the entire document and store it or its hash in DB. when the user opens a document, the app will invoke the getFileAsync function to compare the opening doc with the existing one in the DB to see if it’s the same document. 

The code perfectly works on Office Online. But on other platforms (such as Word desktop), it has problems. The getFileAsync function will return different results even we open the same doc and does not do any changes.  And when I try to exit Word, it will prompt “save” options.
 

This means the document has been modified by getFileAsync function.  Thus I am wondering this is an unknown bug. 
Here is my [test code](https://hanxia-my.sharepoint.com/:u:/g/personal/wyd_hanxia_onmicrosoft_com/EfGUeZ8TVjpEroWoA03_0okBBSwn04a-ejx7e3iiSeRclg?e=puieuC) and the attachment 
![word-addin](https://user-images.githubusercontent.com/44992853/48394539-e75e9180-e74e-11e8-881f-066f9587c806.gif)
shows this issue directly. 

any info is appreciated

**Comment by dmahugh** [comment]
Thanks for reporting this, @isvdevchat, it does indeed sound like a bug to me if the document is modified by calling `getFileAsync` to get the content. I've assigned to my colleague @JuaneloJuanelo so that he can take a look, and we'll follow up when we have more information.

**Comment by JuaneloJuanelo** [comment]
Hello @isvdevchat ! we did an investigation on this issue and here are the results: From what you are describing you are trying to compare the documents just based on the file size (or at least that's the first premise you are using). This is not a reliable way of comparing 2 documents.  The getFileAsync method creates an identical copy of the file (content-wise), however additional metadata is added (Revision IDs on the settings.xml file within the docs OOXML) not necessarily only on save.   So the file size might not be the same and should not be used to verify or disgard if 2 files are the same.  (obviously if the file is many hundreds of KB in difference probably is a good indicator that its not the same).

That been said, what I recommend you to do is to compare the document.xml part of both documents, that will tell you if the 2 docs are the same or not with 100% accuracy.  (you can use body.getOOXML and send the document element to a server to compare it against its counterpart on your DB)

**Comment by ninonextix** [comment]
Hello, @JuaneloJuanelo.

I was the one who contacted the dev support who opened this issue. For clarifications, the add-in calls the getFileAsync() function, converts the output into hash, and compares that hash to the database. May we know why this works on Word Online and not on other platforms? Moreover, we have a question regarding your recommendation for it.

We've tried to invoke the [body.getOOXML](https://docs.microsoft.com/en-us/javascript/api/word/word.body?view=office-js#getooxml--) method from an add-in on Word Online and copied the output from the console.log() part. We did it twice consecutively and compared the two, and have found out that both of them are not the same. They differ on their w:rsidDefault value on the w:document element, and on more parts not included inside the w:document element. We've also found out that the w:document element only contains the text found on the document, not including the images that are inserted. With this, we cannot fully compare the document with the one on the database. Are we missing something in this part?

I've attached both files (the output from the getOOXML method) for a direct comparison:  [XML.zip](https://github.com/OfficeDev/office-js/files/2613723/XML.zip)

Hoping for a positive response on the said matter.
